### DISCUSSION TITLE
Building an asreview plugin

### DISCUSSION THREAD

**Discussion by retorquere** [asreview/asreview#737] [Ideas]
I see that there are asreview plugins for covid datasets, I would like to make a plugin that does something similar for an RSS feed (from a Medline search). Is there documentation on creating asreview plugins?

**Comment by ottomattas** [comment]
We are working on overhauling the related documentation right now. Can you already share something @JTeijema?

**Comment by jteijema** [comment]
New documentation on how to make a dataset extension will be released soon. It will explain how a developer can use the [asreview/template-extension-new-dataset template](https://github.com/asreview/template-extension-new-dataset)!

This repo is online now, so take a look if you want to get ahead of the curve :)

**Comment by Rensvandeschoot** [comment]
@retorquere do let us know if you need more information

**Reply by retorquere** [reply]
Sorry for the late reaction -- I've been swamped in the past few weeks, I hope to get started on a plugin next week.

**Reply by retorquere** [reply]
Is there a way for a plugin to ask for/store configuration data through the asreview app?

**Reply by jteijema** [reply]
Unless I am mistaken (@J535D165 ?), I think that getting and storing config data for a plugin is currently not implemented through the webinterface. That doesn't mean it is not possible, but it might be difficult.

A [subcommand extention ](https://asreview.readthedocs.io/en/latest/API/extension_dev.html#subcommand-extensions)might be an easier way to implement configuration in the meantime.

**Reply by retorquere** [reply]
Can a subcommand extension also be a dataset extension? I want to implement a dataset extension that reads a PubMed search RSS feed, but I need a way for the user to get that feed URL into the extension.

**Reply by J535D165** [reply]
Yes, you can combine both. Put an entry point for the datasets plugin and an entry point for the subcommand in the setup file. 

My suggestion would be to store the RSS config with a subcommand in the `.asreview` folder. Use the dataset plugin to read the RSS config from the `.asreview` folder and create the datasets. See https://github.com/asreview/template-extension-new-dataset/blob/ad67a0c4e0b14fd5c92a5f93717698900f229548/asreviewcontrib/dataset_name/your_dataset.py#L4-L29

For the `.asreview` folder, it is preferred to use this function https://github.com/asreview/asreview/blob/5a5410e3ac833e707ee7563cd932c4a636fdc304/asreview/webapp/utils/paths.py#L20-L34.

I think building this subcommand&datasets extension is quite straightforward. But feel free to drop a message if you need help.
